optimum/neuron/utils/hub_neuronx_cache.py

```diff
@@ -282,11 +349,16 @@ def synchronize_hub_cache(cache_repo_id: Optional[str] = None):
         repo_id (`Optional[str]`, default to None):
             The id of the HuggingFace cache repository, in the form 'org|user/name'.
     """
+    # Not pushing anything if neuron parallel compile.
```

Comment:
I think the check should rather be done in the module calling this. The cache should not know anything about parallel compile.

Reply:
I was thinking the same thing, will make the change.